rootex/framework/systems/script_system.cpp

```diff
@@ -15,30 +15,25 @@ ScriptSystem* ScriptSystem::GetSingleton()
 
 void ScriptSystem::begin()
 {
-	ScriptComponent* scriptComponent = nullptr;
-	for (auto&& component : s_Components[ScriptComponent::s_ID])
+	for (auto&& [id, entity] : EntityFactory::GetSingleton()->getEntities())
 	{
-		scriptComponent = (ScriptComponent*)component;
-		scriptComponent->onBegin();
+		entity->getScript()->evaluateOverrides();
```

Comment:
This won't work well if the script is added to an entity or if the entity is instantiated in the middle of the level. `System::begin()` notifies the beginning of the logic of the system if it exists at initialization time, and not the beginning of the script being added in the world

Reply:
Consider moving this inside `Entity`, whenever the script is attached to it